src/native/Net.h

```diff
@@ -260,11 +267,14 @@ class NetworkManager : public StaticSharedEbb<NetworkManager> {
     };
 
     void ReceiveArp(EthernetHeader& eh, std::unique_ptr<MutIOBuf> buf);
-    void ReceiveIp(EthernetHeader& eh, std::unique_ptr<MutIOBuf> buf);
+    void ReceiveIp(EthernetHeader& eh, std::unique_ptr<MutIOBuf> buf,
+                   uint64_t rxflag = 0);
     void ReceiveIcmp(EthernetHeader& eh, Ipv4Header& ih,
                      std::unique_ptr<MutIOBuf> buf);
-    void ReceiveUdp(Ipv4Header& ih, std::unique_ptr<MutIOBuf> buf);
-    void ReceiveTcp(const Ipv4Header& ih, std::unique_ptr<MutIOBuf> buf);
+    void ReceiveUdp(Ipv4Header& ih, std::unique_ptr<MutIOBuf> buf,
+                    uint64_t rxflag = 0);
+    void ReceiveTcp(const Ipv4Header& ih, std::unique_ptr<MutIOBuf> buf,
+                    uint64_t rxflag = 0);
```

Comment:
What's the advantage of passing rxflag all the way up the network stack?

Reply:
its mainly for hardware checksumming? If the hardware checksums a type of packet, then it'll set a bit in rxflag, so we don't have to do it in network stack